WordPress/Classes/Services/PostService+Likes.swift

```diff
@@ -0,0 +1,68 @@
+extension PostService {
+
+    /**
+     Fetches a list of users that liked the post with the given ID.
+     
+     @param postID  The ID of the post to fetch likes for
+     @param siteID  The ID of the site that contains the post
+     @param success A success block
+     @param failure A failure block
+     */
+    func getLikesFor(postID: NSNumber,
+                     siteID: NSNumber,
+                     success: @escaping (([LikeUser]?) -> Void),
+                     failure: ((Error?) -> Void)? = nil) {
+
+        guard let remote = PostServiceRemoteFactory().restRemoteFor(siteID: siteID, context: managedObjectContext) else {
+            DDLogError("Unable to create a REST remote for posts.")
+            failure?(nil)
+            return
+        }
+
+        remote.getLikesForPostID(postID) { remoteLikeUsers in
+
+            self.deleteExistingUsersFor(postID: postID)
+
+            guard let remoteLikeUsers = remoteLikeUsers,
+                  !remoteLikeUsers.isEmpty else {
+                success(nil)
+                return
+            }
+
+            success(self.createNewUsers(from: remoteLikeUsers))
```

Comment:
A core data best practice would be to handle the deletion and creation of new assets in the background rather than on the main thread. `ContextManager.shared.newDerivedContext`, `context.performBlock` and `ContextManager.shared.saveContext:withCompletionBlock:` would be things to look at.  Also `DispatchQueue.main.async` for calling the success and failure blocks back on the main thread.
The service itself could be created with a derived context (I think it's using the mainContext now), which might simplify things, but it still might make sense to call the success/failure blocks on the main thread from within the service.

Reply:
I'll look into it and give it another go. Thank you!!